File: api-tester/src/main/java/com/revenuecat/apitester/java/CustomerInfoAPI.java
package com.revenuecat.apitester.java;

import android.net.Uri;

import com.revenuecat.purchases.CustomerInfo;
import com.revenuecat.purchases.EntitlementInfos;
import com.revenuecat.purchases.models.Transaction;

import java.util.Date;
import java.util.List;
import java.util.Map;
import java.util.Set;

@SuppressWarnings({"unused", "SpellCheckingInspection"})
final class CustomerInfoAPI {
    static void check(final CustomerInfo customerInfo) {
        final EntitlementInfos entitlementInfo = customerInfo.getEntitlements();
        final Set<String> asubs = customerInfo.getActiveSubscriptions();
        final Set<String> skus = customerInfo.getAllPurchasedSkus();
        final Date led = customerInfo.getLatestExpirationDate();

        final List<Transaction> nst = customerInfo.getNonSubscriptionTransactions();
        final Date opd = customerInfo.getOriginalPurchaseDate();
        final Date rd = customerInfo.getRequestDate();
        final Date fs = customerInfo.getFirstSeen();
        final String oaui = customerInfo.getOriginalAppUserId();
        final Uri mu = customerInfo.getManagementURL();

        final Date eds = customerInfo.getExpirationDateForSku("");
        final Date pds = customerInfo.getPurchaseDateForSku("");
        final Date ede = customerInfo.getExpirationDateForEntitlement("");
        final Date pde = customerInfo.getPurchaseDateForEntitlement("");
        final Map<String, Date> allExpirationDatesByProduct = customerInfo.getAllExpirationDatesByProduct();
        final Map<String, Date> allPurchaseDatesByProduct = customerInfo.getAllPurchaseDatesByProduct();
    }
}

File: api-tester/src/main/java/com/revenuecat/apitester/kotlin/CustomerInfoAPI.kt
package com.revenuecat.apitester.kotlin

import android.net.Uri
import com.revenuecat.purchases.CustomerInfo
import com.revenuecat.purchases.EntitlementInfos
import com.revenuecat.purchases.models.Transaction
import java.util.Date

@Suppress("unused", "UNUSED_VARIABLE")
private class CustomerInfoAPI {
    fun check(customerInfo: CustomerInfo) {
        with(customerInfo) {
            val entitlementInfo: EntitlementInfos = entitlements
            val asubs = activeSubscriptions
            val skus: Set<String> = allPurchasedSkus
            val productIds: Set<String> = allPurchasedProductIds
            val led: Date? = latestExpirationDate
            val nst: List<Transaction> = nonSubscriptionTransactions
            val opd: Date? = originalPurchaseDate
            val rd: Date = requestDate
            val fs: Date = firstSeen
            val oaui: String = originalAppUserId
            val mu: Uri? = managementURL
            val eds: Date? = getExpirationDateForSku("")
            val edpi: Date? = getExpirationDateForProductId("")
            val pds: Date? = getPurchaseDateForSku("")
            val pdpi: Date? = getPurchaseDateForProductId("")
            val ede: Date? = getExpirationDateForEntitlement("")
            val pde: Date? = getPurchaseDateForEntitlement("")
            val allExpirationDatesByProduct: Map<String, Date?> = allExpirationDatesByProduct
            val allPurchaseDatesByProduct: Map<String, Date?> = allPurchaseDatesByProduct
        }
    }
}
